### DISCUSSION TITLE
Function to switch between firmware versions or v2 firmware as default?

### DISCUSSION THREAD

**Discussion by FeuerSturm** [JAndrassy/WiFiEspAT#84] [Q&A] [answered]
Hey @JAndrassy ,
would it be possible to add a function to easily switch the firmware version on a per sketch base?
Or use FW v2 as default instead of v1?
I need to require your great lib for my project that a lot of rookies will use that even commenting a single line in one file might be asked to much for.

Something simple like "setFWversion(2)" maybe?

Thanks in advance!

**Comment by JAndrassy** [comment]
there is no space for two versions of the firmware in the esp8266

**Reply by FeuerSturm** [reply]
Hi @JAndrassy ,

thanks for the quick reply, but I guess I didn't make myself clear enough what I am asking for:
I am asking for an "on-the-fly" way to change what AT firmware version your lib expects to be running on the MCU,
so instead of commenting/uncommenting `#define WIFIESPAT1` before compiling a project, it would be great if we could
just change that in the sketch itself.

**Reply by JAndrassy** [reply]
it would make the library much larger. it would take more flash memory

**Comment by FeuerSturm** [comment, marked as answer]
Alright, thanks for the reply! :+1:
